olm/operator-registry.md

```diff
@@ -0,0 +1,291 @@
+2. Run the manifest image with that binary that writes to a configmap in a well known location
+3. Read the configmap data
+4. On return kill the pod
+5. Delete the configmap
+
+pod spec:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+  containers:
+  - name: operator-manifest
+    image: quay.io/community-operators/foo@sha256:abc123
+    command: ['operator-registry', 'serve' '-manifestdir "./manifests/"']
+  initContainers:
+  - name: init-manifest
+    image: quay.io/operator-framework/init-operator-manifest:latest
+    command: ['cp', 'operator-registry', '.']
+```
+
+We will also need to update operator-registry to include a new `serve` command that knows how to parse the manifest/ folder in the operator bundle image. This command will write to a configmap that OLM will use to get the bundle data.
+
+*DB Versioning*
+
+Since we are now calling this database schema a first class releasable object, we also need to be able to make changes to the schema over time. Along with that, we will need to be able to migrate to newer schema versions over time. To get started with that, we will update the existing operator-registry database to include a table that defines the current version of the database schema.
+
+*Make CatalogSource type for this image*
+
+Finally, once these operator registry images are servable, we will update the catalog source to be able to download these images and serve them.
+
+Additionally, we will update the spec to create a polling field that allows the index to be updated automatically on a configurable poll timer.
+
+```
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: example-catalog
+  namespace: olm
+spec:
+  sourceType: grpc
+  image: quay.io/operator-framework/community-operators:
+  pollTimer: 1m
+```
+
+### Risks and Mitigations
+
+Concern:
+
+Upstream community cannot come up with agreement about what an operator bundle image looks like. 
+
+Mitigation:
+
+In that event, we will do a subset of this work. Instead of the operator-registry taking an image, it will take a local directory that has a set of yaml for the bundle. Additionally, we will not add the imagePath optimization to the registry schema.
+
+## Design Details
+
+### Test Plan
+
+Update operator-registry tests to ensure new APIs and DB update commands work as expected.
+Additionally, add e2e tests with mocked index containers to ensure that those workflows work as expected.
+
+### Graduation Criteria
+
+##### Dev Preview -> GA
+
+Ability to build indexes
+Ability to apply them to a cluster
+Ability to drive UI/UX with indexes
+
+##### Removing a deprecated feature
+
+n/a
+
+### Upgrade / Downgrade Strategy
+
+Net new feature for now. In a future enhancement, we will attempt to enhance this feature further as a way of replacing default operator sources with default index catalog sources.
+
+### Version Skew Strategy
+
+n/a
```

Comment:
This seems like a good place to put our versioning strategy 😄

Reply:
+1